sql/sql_table.cc

```diff
@@ -8854,13 +8854,17 @@ bool mysql_alter_table(THD *thd, const char *new_db, const char *new_name,
 
   if (versioned)
   {
-    if (create_info->db_type && table->s->db_type() != create_info->db_type &&
-        (table->file->native_versioned() ||
-         ha_check_storage_engine_flag(create_info->db_type,
-                                      HTON_NATIVE_SYS_VERSIONING)))
+    if (handlerton *hton1= create_info->db_type)
```

Comment:
This should bring warning about suggested parentheses.

Reply:
I don't see such warning neither with gcc nor with clang.